src/sage/data_structures/stream.py

```diff
@@ -381,21 +381,45 @@ def __getitem__(self, n):
 
         if self._is_sparse:
             try:
-                c = self._cache[n]
+                return self._cache[n]
             except KeyError:
```

Comment:
I would rather not, since this is absolutely time critical, and I doubt the benefit of moving in this particular case.   If I see the return here, it is completely clear that nothing else has to be done if the thing is in the cache.  Otherwise I'd have to follow the indentation of the except block and after it, which is currently irrelevant.

Reply:
I don't follow your logic here. Either way you still have to follow what is after. Essentially this is the same as
```python
if n in self._cache:  # the "special" case
    return self._cache[n]
# continue on with the general case
```
albeit a faster version when it is in the cache (although slower when it is not). By having the stuff currently in the exception block moved outside, it makes it seem like it could potentially flow further down. You're not doing anything with the error message nor anything specific with regards to the type of error raised.

I am not going to push too much on this as it is a fairly trivial point, but it breaks a bit what I learned as good (general) Python error handling: Put as little as possible within `try`-`except` blocks and limit the amount of logic flow around such blocks.

Reply:
I just realized that I completely misunderstood your suggestion.  I thought that you wanted me to remove the `return` after the `try:` statement.  Will commit shortly.